README.md

```diff
@@ -1 +1,84 @@
 # hof-lookup-replacer
+
+A Node.js script for replacing db lookup tables as a scheduled job configurable to different HOF form services.
+
+The script retrieves a CSV formatted file from a given location, validates against given rules and then replaces a lookup table using the CSV rows. On success or failure the script will send appropriate notification messages.
+
+This service uses either [pg-promise](https://vitaly-t.github.io/pg-promise/index.html) or [knex.js](https://knexjs.org/) to handle db connection and query building. You can configure the client and model used using environment variables.
+
+## Script steps
+
+1. Scheduled job runs with service specific environment.
+2. Connect to database and retrieve the latest uploaded data's S3/file-vault URL from an SQL database.
+3. Authenticate with the service's file-vault.
+4. Use the data URL to bring the CSV file data into the application.
+5. Validate the CSV rows against rules given for the service. Log, note and remove erroring rows.
+6. Replace the SQL db's lookup table with the validated rows from CSV.
+7. Prepare a and send a notification email of pass or fail including erroring rows noted in validation.
+8. Close connection. Job complete.
+
+## Configuration
+
+The `config.js` file turns most of the service's environment variables into an exported object that can be referred to throughout the app. Some defaults are configured here.
+
+### Database
+
+hof-lookup-replacer aims to be database agnostic - the main script should run given different database models. The model chosen can be configured by the `DB_MODEL` environment variable being one of the filenames of the models provided (e.g. `DB_MODEL=knex-postgres-model`). The default is pgp-model which uses pg-promise.
```

Comment:
why pg-promise?

Reply:
Defaults are no longer set so this line is removed and pg-promise is no longer a default. pg-promise is reportedly a more efficient tool to do batch inserts if knex performs slowly.

Reply:
nice one